pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -148,6 +149,63 @@ func (r *hibernationReconciler) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
+	// Check for the hibernate-after annotation, and if the cluster has been in running state for longer than this duration, put it to sleep.
+	hibernateAfter, ok := cd.Annotations[constants.HibernateAfterAnnotation]
+	if ok && cd.Spec.PowerState != hivev1.HibernatingClusterPowerState {
```

Comment:
Also add a check for whether the cluster can handle hibernation. If the cluster cannot handle hibernation, then it is awkward to have Hive setting the power state to Hibernating.

Reply:
Went to implement this but then realized we should probably surface the problem if you set this annotation on a cluster that can't handle it. We can't really log that, nobody will ever see it, it should be a condition. If we leave it, we get that condition for free, when it procs we'll move a cluster to hibernating and it'll get a condition explaining why it won't work. Granted it's delayed until it's time to hibernate but nobody's going to look until something is wrong regardless. I think the current form might be better.

Reply:
This is an example of where the lines getting blurred between the roles of spec and annotation. If something in the annotation is enough to surface a condition on the object, then it is certainly worthy of being in the spec proper. If does nobody any good to continue to put things like this into annotations where it only serves to make it harder for the user to find information about and use.

Reply:
If the feature is for Hive developers, then the Hive developer should know enough about the annotation to not need a condition telling them that they did something unsupported. If the feature is for Hive users, then it should be in the spec.